### ISSUE TITLE
Sign in with Linkedin throws exception if the user hits cancel from Linkedin's website

### ISSUE THREAD

**Issue by felixSabatie** [aspnet-contrib/AspNet.Security.OAuth.Providers#480]
**Describe the bug**
Similar to #424 , if a user cancels the sign in request an exception is thrown instead of the user going to the AccessDeniedPath.

**Steps To reproduce**
Set up sign in with Linkedin. On Linkedin's website, cancel at the sign in step or at the authorize step.

**Expected behaviour**
The user should be redirected to the url defined in the AccessDeniedPath.

**Actual behaviour**
An exception is thrown, user_cancelled_login if the user cancels at the login step or user_cancelled_authorize if the user cancels at the application authorization step on Linkedin's website.

```
System.Exception: user_cancelled_login;Description=The user cancelled LinkedIn login

System.Exception: An error was encountered while handling the remote login.
 ---> System.Exception: user_cancelled_login;Description=The user cancelled LinkedIn login
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler`1.HandleRequestAsync()
   at IdentityServer4.Hosting.FederatedSignOut.AuthenticationRequestHandlerWrapper.HandleRequestAsync()
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at IdentityServer4.Hosting.BaseUrlMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**System information:**
 - OS: Windows 10
 - Library Version AspNet.Security.OAuth.LinkedIn 3.1.3
 - .NET version 3.1.301

**Comment by martincostello** [comment]
Thanks @felixSabatie - I'll have a look into fixing this for the next release.

**Comment by kevinchalet** [comment]
How many non-standard error codes do they have in stock? 🤣

**Comment by martincostello** [comment]
Have opened #481 with a proposed fix.

As this is the second time we've had to override this, I'm going to open an issue against the dotnet/aspnetcore repo asking if there's any worth in adding functionality as part of .NET to allow additional error codes meaning denied to be user-specified as part of the OAuthOptions. I'm not sure they'd take it, but it's worth asking the question.

**Comment by martincostello** [comment]
Have raised https://github.com/dotnet/aspnetcore/issues/26540.

**Comment by martincostello** [comment]
@felixSabatie If you could try out the [`3.1.4-preview.20503.40`](https://www.myget.org/feed/aspnet-contrib/package/nuget/AspNet.Security.OAuth.LinkedIn/3.1.4-preview.20503.40) package from our MyGet feed and confirm it resolves your issue, I can push out a 3.1.4 release to NuGet.org containing the fix.

**Comment by felixSabatie** [comment]
@martincostello It works great with the update, thanks for the quick implementation 😃

**Comment by martincostello** [comment]
Great, thanks for confirming 😃

I'll look at pushing a release with the fix to NuGet within the next day or so.

**Comment by martincostello** [comment]
This fix is now available in version [3.1.4](https://www.nuget.org/packages/AspNet.Security.OAuth.LinkedIn/3.1.4) of the provider - thanks again for reporting the problem!
